bootstrap/00_once_cicd_slack_build_messages.yaml

```diff
@@ -0,0 +1,99 @@
+AWSTemplateFormatVersion: '2010-09-09'
+Description: Configurazioni necessarie per scrivere log API GateWay su CloudWatch
+
+Parameters:
+  SlackInfo:
+    Type: String
+    Description: <slack workspace id>:<slack channel id>
+    Default: TQSBH3ZS4:C028UC74ZEV 
+
+Resources:
+  
+  # BUILD FAILURE SLACK NOTIFICATION TOPIC
+  ########################################
+  BuildNotificationSNSTopic:
+    Type: AWS::SNS::Topic
+  TopicPolicy:
+    Type: AWS::SNS::TopicPolicy
+    Properties:
+      PolicyDocument:
+        Statement:
+          - Sid: AllowServices
+            Effect: Allow
+            Principal:
+              Service:
+                - "codestar-notifications.amazonaws.com"
+            Action:
+              - 'sns:Publish'
+            Resource:
+              - !Ref BuildNotificationSNSTopic
+          - Sid: AllowAWS
+            Effect: Allow
+            Principal:
+              AWS: "*"
+            Action:
+              - "SNS:GetTopicAttributes"
+              - "SNS:SetTopicAttributes"
+              - "SNS:AddPermission"
+              - "SNS:RemovePermission"
+              - "SNS:DeleteTopic"
+              - "SNS:Subscribe"
+              - "SNS:ListSubscriptionsByTopic"
+              - "SNS:Publish"
+              - "SNS:Receive"
+            Resource:
+              - !Ref BuildNotificationSNSTopic
+      Topics:
+        - !Ref BuildNotificationSNSTopic
+
+  BuildNotificationChatbotIAMRole:
+    Type: AWS::IAM::Role
+    Properties:
+      Description: Role for build notifications chatbot
+      AssumeRolePolicyDocument:
+        Version: '2012-10-17'
+        Statement:
+        - Effect: Allow
+          Principal:
+            Service:
+            - chatbot.amazonaws.com
+          Action:
+          - sts:AssumeRole
+      Path: "/service_role/"
+      ManagedPolicyArns:
+      - arn:aws:iam::aws:policy/CloudWatchReadOnlyAccess
+  BuildNotificationChatbot:
+    Type: AWS::Chatbot::SlackChannelConfiguration
+    Properties:
+      ConfigurationName: !Sub 'notifications-chatbot'
+      IamRoleArn: 
+        Fn::GetAtt: [ BuildNotificationChatbotIAMRole, Arn ]
+      LoggingLevel: INFO
+      SlackChannelId:
+          Fn::Select:
+          - '1'
+          - Fn::Split:
+            - ":"
+            - Ref: SlackInfo
+      SlackWorkspaceId:
+          Fn::Select:
+          - '0'
+          - Fn::Split:
+            - ":"
+            - Ref: SlackInfo
+      SnsTopicArns:
+        - Ref: BuildNotificationSNSTopic
+  BuildNotificationTopicChatbot:
+    Type: AWS::SNS::Subscription
+    Properties:
+      TopicArn: 
+        Ref: BuildNotificationSNSTopic
+      Endpoint: "https://global.sns-api.chatbot.amazonaws.com"
+      Protocol: https
+
+Outputs:
+  BuildNotificationSNSTopicArn:
+    Description: ARN of SNS Topic for build notification
+    Value: !Ref BuildNotificationSNSTopic
+    Export:
+      Name: !Sub "build-msg-topic"
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Fn::Sub isn&#39;t needed because there are no variables at Outputs/BuildNotificationSNSTopicArn/Export/Name/Fn::Sub

Reply:
rimosso